### DISCUSSION TITLE
Has anyone successfully compiled Kepler.gl on Ubuntu?

### DISCUSSION THREAD

**Discussion by 0xd5dc** [keplergl/kepler.gl#2624] [Q&A]
Has anyone successfully set up this project? If so, could you please share some tips or a step-by-step guide for getting it up and running? 

My Kepler.gl Project Setup Issues
I've encountered significant challenges while attempting to set up the Kepler.gl project across various environments. Here's a summary of my experience:
Testing Existing Demos
I tested the project using Node.js version 18 and later versions on multiple operating systems:

    Windows
    Ubuntu VM
    macOS

Unfortunately, only, on macOS, two out of the eight example demos successfully compiled and rendered the Kepler UI as expected.
Attempting a Fresh Build
I also tried to build this project from scratch. However, I ran into numerous dependency issues that prevented me from even starting the project.

**Comment by ibgreen** [comment]
Kepler maintainers all work on MacOS. 

We would accept reasonable small fixes for building on other operating systems. 

While more extensive support would be great, it is currently beyond our available resources and priorities to provide active support for developing on non-Mac environments, unless someone actively working on one of these environments steps up and helps us support it.

We could certainly clarify this in the docs.

**Comment by ibgreen** [comment]
When it comes to the examples the situation is different. They should at least be fixed on MacOS (or removed if no longer valid / easily fixable, we should not have broken examples).

kepler is currently going through a big, long overdue overhaul of the build environment, moving to latest esbuild, yarn etc.

It may break some things in the short term, but longer term is should put the repo in a much better state, so bear with us for a little while.

(Of course, it is also possible that by moving to the latest tooling, it becomes easier to support non-MacOS envs, but that is pure speculation on my part.)

**Reply by 0xd5dc** [reply]
Thank you for the explanation. I'm working on a project to visualize a large dataset (100GB) and have successfully rendered a small subset on macOS. However, I'm facing challenges with rendering a larger portion of the data on macOS. I plan to move the project to my production server, and I assume it's possible, but I’ll need to explore the options myself.

**Comment by ibgreen** [comment]
The limitation we are talking about relates to building kepler.gl, not running kepler. Once kepler.gl is built, you should be able to serve the bundle from any environment. Kepler runs in the browser, not on your server, unless you have a custom setup. So just build your bundle on MacOS and serve it from wherever?
